theano/gpuarray/dnn.py

```diff
@@ -3717,6 +3721,89 @@ def local_dnn_argmax(op, ctx_name, inputs, outputs):
     return [as_gpuarray_variable(arg.astype('int64'), ctx_name)]
 
 
+@register_opt('cudnn', 'fast_compile')
+@op_lifter([TransformerGrid])
+@register_opt2([TransformerGrid], 'fast_compile', 'cudnn')
+def local_dnn_spatialtf_grid(op, ctx_name, inputs, outputs):
+    if not dnn_available(ctx_name):
+        return
+
+    theta, out_dims = inputs
+
+    if theta.ndim != 3:
+        return
+
+    if out_dims.ndim != 4:
+        return
+
+    grid = GpuDnnTransformerGrid()(theta, out_dims)
+
+    return [grid]
+
+
+@register_opt('cudnn', 'fast_compile')
+@op_lifter([TransformerSampler])
+@register_opt2([TransformerSampler], 'fast_compile', 'cudnn')
+def local_dnn_spatialtf_sampler(op, ctx_name, inputs, outputs):
+    if not dnn_available(ctx_name):
+        return
+
+    inp, _grid = inputs
+    grid = _grid.dimshuffle(1, 2, 3, 0)
+
+    if inp.ndim != 4:
+        return
+
+    if grid.ndim != 4:
+        return
+
+    sampler = GpuDnnTransformerSampler()(inp, grid)
```

Comment:
CPU transformer sampler has a border_mode attribute, but you don't check it here. Can all CPU border modes be transferred to GPU ? I don't think so, isn'it ?

Reply:
I don't think they can be transparently transferred to the GPU, and cuDNN does not provide such a parameter, nor does it is documented how the borders are treated in the implementation. Comparing the implementation on the CPU using the nearest mode, computes the same outputs, I will test with other modes and report back.

Reply:
Ok. If nearest mode is the only one supported in cuDNN, then you should check it here and optimize only if op.border_mode == 'nearest'.

Reply:
I've checked with every mode, comparing the output with GPU, and they all pass here. Can you also check it?